src/main/java/org/quiltmc/loader/impl/transformer/ChasmInvoker.java

```diff
@@ -0,0 +1,243 @@
+						try {
+							return Files.readAllBytes(path2);
+						} catch (IOException e) {
+							return null;
+						}
+					}
+				}
+				return null;
+			}
+
+			@Override
+			public boolean isInterface(String className) {
+				ClassReader cr = new ClassReader(readFile(LoaderUtil.getClassFileName(className)));
+				ClassInfoGrabber namer = new ClassInfoGrabber();
+				cr.accept(namer, ClassReader.SKIP_CODE | ClassReader.SKIP_DEBUG | ClassReader.SKIP_FRAMES);
+				return (Opcodes.ACC_INTERFACE & namer.access) != 0;
+			}
+
+			@Override
+			public boolean isAssignable(String leftClass, String rightClass) {
+				if ("java/lang/Object".equals(leftClass)) {
+					return true;
+				}
+
+				byte[] file = readFile(LoaderUtil.getClassFileName(rightClass));
+				if (file == null) {
+					return false;
+				}
+
+				ClassReader cr = new ClassReader(file);
+				ClassInfoGrabber namer = new ClassInfoGrabber();
+				cr.accept(namer, ClassReader.SKIP_CODE | ClassReader.SKIP_DEBUG | ClassReader.SKIP_FRAMES);
+
+				if (namer.superName.equals(leftClass)) {
+					return true;
+				}
+
+				for (String itf : namer.interfaces) {
+					if (itf.equals(leftClass)) {
+						return true;
+					}
+				}
+
+				if (isAssignable(leftClass, namer.superName)) {
+					return true;
+				}
+
+				for (String itf : namer.interfaces) {
+					if (isAssignable(leftClass, itf)) {
+						return true;
+					}
+				}
+
+				return false;
+			}
+
+			@Override
+			public String getSuperClass(String className) {
+				ClassReader cr = new ClassReader(readFile(LoaderUtil.getClassFileName(className)));
+				ClassInfoGrabber namer = new ClassInfoGrabber();
+				cr.accept(namer, ClassReader.SKIP_CODE | ClassReader.SKIP_DEBUG | ClassReader.SKIP_FRAMES);
+				return namer.superName;
+			}
+		});
+
+		for (ModLoadOption mod : modList) {
+			Path modPath = root.resolve(mod.id());
+			if (!Files.exists(modPath)) {
+				continue;
+			}
+			Path chasmRoot = modPath.resolve("chasm_transformers");
+			Files.walkFileTree(modPath, new SimpleFileVisitor<Path>() {
+				@Override
+				public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+					if (file.getFileName().toString().endsWith(".class")) {
+						byte[] bytes = Files.readAllBytes(file);
+						Metadata meta = new Metadata();
+						meta.put(QuiltMetadata.class, new QuiltMetadata(mod));
+						chasm.addClass(new ClassData(bytes, meta));
+					} else if (file.startsWith(chasmRoot) && file.getFileName().toString().endsWith(".chasm")) {
+						String chasmId = mod.id() + ":" + chasmRoot.relativize(chasmRoot).toString();
+						Log.info(LogCategory.GENERAL, "Found chasm transformer: '" + chasmId + "'");
+						Node node = Node.parse(file);
+						Transformer transformer = new ChasmLangTransformer(chasmId, node, chasm.getContext());
+						chasm.addTransformer(transformer);
+					}
+					return FileVisitResult.CONTINUE;
+				}
+			});
+		}
+
+		List<ClassData> changed = chasm.process(true);
```

Comment:
Just letting you know that I opened an issue here: https://github.com/QuiltMC/chasm/issues/96
This will be a breaking change to this part of the code